src/components/ListView.jsx

```diff
@@ -50,21 +48,54 @@ function ListView() {
   //
   const handleCheckboxChange = async (e) => {
     const itemId = e.target.name;
-    //create time variable to save time when user checked the box as purchasedTime
-    const datePurchased = Date.now();
-    //if user want to uncheck the item it can be done and purchasedDate is set to null again
+    // grabbing the specific item from state that is clicked because we will be updating its properties
+    const itemToUpdate = items.find((item) => itemId === item.itemName);
+
+    let daysSinceLastTransaction;
+
+    // if user checks a box, itemToUpdate is taken through this flow
     if (e.target.checked) {
-      setUpdateToDb(localToken, itemId, 'purchasedDate', datePurchased);
+      // check to see if the item has never been purchased before (purchasedDate is set to null by default)
+      if (!itemToUpdate.purchasedDate) {
```

Comment:
what if we use (itemToUpdate.totalPurchases < 2) cause we know that if totalPurchases < 2 calculateEstimate will return daysSinceLastTransaction anyway

Reply:
thanks a lot for this comment, I'd never think about this